maitake/src/scheduler.rs

```diff
@@ -21,16 +21,22 @@ pub struct Tick {
     pub has_remaining: bool,
 }
 
-pub trait Schedule: Sized + Clone {
-    fn schedule(&self, task: TaskRef);
+use sealed::Schedule as _;
+pub(crate) mod sealed {
+    use crate::task::TaskRef;
 
-    /// Returns a new [task `Builder`] for configuring tasks prior to spawning
-    /// them on this scheduler.
-    ///
-    /// [task `Builder`]: task::Builder
-    #[must_use]
-    fn build_task<'a>(&self) -> task::Builder<'a, Self> {
-        task::Builder::new(self.clone())
+
+    pub trait Schedule: Sized + Clone {
```

Comment:
I'm not sure if this is really the right way to seal the trait. I think user code _does_  need to be able to reference the trait name, so that it can write code that is generic over `S: Schedule`, or potentially also use the trait's methods. It just can't be _implemented_ in user code.

In order to ensure that, I think the trait definition of `Schedule` needs to be in the public module, but it needs to require a `Sealed` marker trait that is only visible internally. Like:

```rust
pub trait Schedule: sealed::Sealed + Sized + Clone {
     // ...
}

mod sealed {
    pub trait Sealed {}
}
```